chorus_lib/src/core.rs

```diff
@@ -393,9 +378,9 @@ impl<L1: ChoreographyLocation, B: Transport> Projector<L1, B> {
                 }
             }
 
-            fn call<R, M, C: Choreography<R, L = M>, Index>(&self, choreo: C) -> R
+            fn call<R, M, C: Choreography<R, L = M>>(&self, choreo: C) -> R
             where
-                M: HList + Equal<L, Index>,
+                M: HList,
```

Comment:
When we `call` a choreography, the location set of the sub-choreography must be a subset of that of the outer choreography.

```suggestion
                M: HList + Subset<L, Index>,
```